compiler/circle-mpqsolver/src/bisection/VISQErrorApproximator.cpp

```diff
@@ -17,23 +17,45 @@
 #include "VISQErrorApproximator.h"
 
 #include <fstream>
+#include <json.h>
 
 using namespace mpqsolver::bisection;
 
 void VISQErrorApproximator::init(const std::string &visq_data_path)
 {
   // read file
   std::ifstream file(visq_data_path);
-  if (!init(file))
+  try
+  {
+    init(file);
+  }
+  catch (std::runtime_error &)
   {
     throw std::runtime_error("Invalid visq file " + visq_data_path);
   }
 }
 
-bool VISQErrorApproximator::init(std::istream &)
+void VISQErrorApproximator::init(std::istream &visq_stream)
```

Comment:
I don't understand why change bool to void in this change.

Reply:
> I don't understand why change bool to void in this change.

Now it throws to show the reason of failure.

Reply:
commit message is `This commit implements visq parsing.` which I don't know the relation with changing this method prototype. If you want to change this, split to another PR.

Reply:
ok. 
@jinevening i'll post another PR with corrected exceptions.